pkg/kv/store_message.go

```diff
@@ -64,8 +64,66 @@ func (s *StoreMessage) Delete(ctx context.Context, path string) error {
 	return err
 }
 
-// TODO(niro): implement Scan when required
+// Scan returns a prefix iterator which returns entries in deserialized format. Scan in store message implementation is msg specific
+// and therefore must be PREFIX bound
+func (s *StoreMessage) Scan(ctx context.Context, msg protoreflect.Message, prefix string) (*MessageIterator, error) {
+	return NewMessageIterator(ctx, s.Store, msg, prefix)
+}
 
 func (s *StoreMessage) Close() {
 	s.Store.Close()
 }
+
+type MessageIterator struct {
+	itr EntriesIterator
+	msg protoreflect.Message
+	err error
+}
+
+type MessageEntry struct {
```

Comment:
You need a `predicate` as well.

Reply:
Can you explain?

Reply:
Look at #3408

Reply:
I would not like to rely on currently WIP PRs. I prefer addressing it only when the changes are committed